lib/src/transformer/impl_generation.dart

```diff
@@ -296,7 +296,7 @@ class ImplGenerator {
             !member.isSynthetic &&
             member.isAbstract &&
             member.name.name == name &&
-            member.returnType?.name?.name == type
+            member.returnType == type
```

Comment:
`member.returnType` is a `TypeAnnotation`, and `type` is a `String`... Not sure how this works.

For better future-proofing, can we change this to the following?
```dart
member.returnType?.toSource() == type
```